pkg/identityserver/api_key_utils.go

```diff
@@ -43,11 +47,25 @@ func GenerateAPIKey(ctx context.Context, name string, rights ...ttnpb.Right) (ke
 	if err != nil {
 		return nil, "", err
 	}
-	key = &ttnpb.APIKey{
-		ID:     generatedID,
-		Key:    hashedKey,
-		Name:   name,
-		Rights: rights,
+	// Check that expiry date passed is not a zero time or nil value
+	if expiresAt != nil && !expiresAt.IsZero() {
+		if expiresAt.Before(time.Now()) { // Check if an expired date is passed.
+			return nil, "", errExpiryDateInPast
+		}
```

Comment:
I don't think this check belongs here. Maybe better to add a `(validate.rules).timestamp.gt_now = true` to the `.proto` files so that it's validated that way?